share/wake/lib/system/path.wake

```diff
@@ -117,15 +117,15 @@ export def isPathDir (path: Path): Boolean =
 
 # Paths differ from Strings in that they have been hashed; their content is frozen
 data Path =
-    Path (name: String)
+    Path (name: String) (hash: String)
 from wake export type Path
 
 export def getPathName (path: Path): String =
-    def Path name = path
+    def Path name _ = path
     name
 
 export def getPathParent (path: Path): Path =
-    Path (simplify "{getPathName path}/..")
+    Path (simplify "{getPathName path}/..") path.getPathHash
```

Comment:
Whatever we decide on, we obviously don't want to use this as-is.

Reply:
Yeah I agree. That's why I left the question about this. I think keeping with directories being all zeros is good. We do not want to hash the contents at all.

Reply:
Definitely not contents.  The only reason I suggest factoring in permissions is because we currently consider a directory from `mkdir` as incompatible with one from `mkdirIn` with a non-0755 mode ("file output by multiple jobs") and it seems like it *might* help virtualization to distinguish those two results, but then again we'd have the same issue with `write`/`writeIn`, and don't do anything with permissions there either.

Reply:
For this change we want to keep the semantics the same. The goal of this change is to change no semantics anywhere. We can discuss other hashing changes elsewhere.

Reply:
I wasn't expecting the hash changes to be part of this PR.  Since we were talking about directory hashing I went off on a bit of a tangent, but it probably does make sense to discuss that somewhere other than buried in the comments of an otherwise-unrelated PR.